org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/wizards/AbapGitWizardSwitchBranch.java

```diff
@@ -0,0 +1,166 @@
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.Assert;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jface.dialogs.DialogPage;
+import org.eclipse.jface.dialogs.IPageChangingListener;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.jface.dialogs.PageChangingEvent;
+import org.eclipse.jface.operation.IRunnableWithProgress;
+import org.eclipse.jface.wizard.IWizardContainer;
+import org.eclipse.jface.wizard.Wizard;
+import org.eclipse.jface.wizard.WizardDialog;
+import org.eclipse.jface.wizard.WizardPage;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.plugin.AbstractUIPlugin;
+
+import com.sap.adt.tools.core.model.adtcore.IAdtObjectReference;
+import com.sap.adt.tools.core.ui.packages.AdtPackageServiceUIFactory;
+import com.sap.adt.tools.core.ui.packages.IAdtPackageServiceUI;
+
+public class AbapGitWizardSwitchBranch extends Wizard {
+
+	private final IProject project;
+	final CloneData cloneData;
+	public IRepository selRepoData;
+	private final String destination;
+	private PageChangeListener pageChangeListener;
+	AbapGitWizardPageRepositoryAndCredentials pageCredentials;
+	AbapGitWizardPageSwitchBranchAndPackage pageBranchAndPackage;
+
+	public AbapGitWizardSwitchBranch(IProject project, IRepository selRepo, String destination) {
+		this.project = project;
+		this.cloneData = new CloneData();
+		this.destination = destination;
+		this.selRepoData = selRepo;
+		this.cloneData.url = selRepo.getUrl();
+		this.cloneData.branch = selRepo.getBranchName();
+		getPackageAndRepoType();
+
+		setWindowTitle(Messages.AbapGitWizardSwitch_branch_wizard_title);
+		setNeedsProgressMonitor(true);
+		setDefaultPageImageDescriptor(
+				AbstractUIPlugin.imageDescriptorFromPlugin(AbapGitUIPlugin.PLUGIN_ID, "icons/wizban/abapGit_import_wizban.png")); //$NON-NLS-1$
+	}
+
+	private void getRepositoryAccessMode() {
+		//Get external repo info for repository type (public / private)
+		IExternalRepositoryInfoService externalRepoInfoService = RepositoryServiceFactory
+				.createExternalRepositoryInfoService(AbapGitWizardSwitchBranch.this.destination, null);
+		AbapGitWizardSwitchBranch.this.cloneData.externalRepoInfo = externalRepoInfoService
+				.getExternalRepositoryInfo(AbapGitWizardSwitchBranch.this.selRepoData.getUrl(), "", "", null); //$NON-NLS-1$ //$NON-NLS-2$
+	}
+
+	private void getPackageRef(String packageName, IProgressMonitor monitor) {
+		IAdtPackageServiceUI packageServiceUI = AdtPackageServiceUIFactory.getOrCreateAdtPackageServiceUI();
+		if (packageServiceUI.packageExists(AbapGitWizardSwitchBranch.this.destination, packageName, monitor)) {
+			List<IAdtObjectReference> packageRefs = packageServiceUI.find(AbapGitWizardSwitchBranch.this.destination, packageName, monitor);
+			AbapGitWizardSwitchBranch.this.cloneData.packageRef = packageRefs.stream().findFirst().orElse(null);
+		}
+	}
```

Comment:
- From here itself we can throw the exception if package does not exist
- Also the progress monitor is not needed much here as the wizard is not yet started. So we can pass a null progress monitor

Reply:
```suggestion
	private void getPackageRef(String packageName) throws PackageRefNotFoundException {
		IAdtPackageServiceUI packageServiceUI = AdtPackageServiceUIFactory.getOrCreateAdtPackageServiceUI();
		if (packageServiceUI.packageExists(AbapGitWizardSwitchBranch.this.destination, packageName, new NullProgressMonitor())) {
			List<IAdtObjectReference> packageRefs = packageServiceUI.find(AbapGitWizardSwitchBranch.this.destination, packageName,
					new NullProgressMonitor());
			AbapGitWizardSwitchBranch.this.cloneData.packageRef = packageRefs.stream().findFirst().orElse(null);
		}
		else {
			throw new PackageRefNotFoundException("Package not found"); //$NON-NLS-1$
		}
	}
```